word_language_model/main.py

```diff
@@ -58,7 +58,7 @@
 if torch.cuda.is_available():
     if not args.cuda:
         print("WARNING: You have a CUDA device, so you should probably run with --cuda.")
-if torch.backends.mps.is_available():
+if ('mps' in dir(torch.backends)) and torch.backends.mps.is_available():
```

Comment:
nit: I think using `hasattr` is cleaner
```suggestion
if hasattr(torch.backends, "mps") and torch.backends.mps.is_available():
```